spec/service_actor/arguments_validator_spec.rb

```diff
@@ -40,4 +40,47 @@
         )
     end
   end
+
+  describe ".validate_default_value" do
+    include_context "with mocked `Kernel.warn` method"
+
+    [[], {}, {a: 1}, {a: []}.freeze, Struct.new(:a).new].each do |mutable_value|
+      context "when `Ractor` API is supported" do
+        include_context "skip on MRI 2.7"
+
+        it "emits a warning if default value `#{mutable_value}` is mutable" do
+          described_class.validate_default_value(mutable_value, origin_type: :input, origin_name: :value, actor: "actor_name")
+
+          expect(Kernel).to have_received(:warn)
+            .with("DEPRECATED: Actor `actor_name` has input `value` with default which is not a Proc or an immutable object.")
+            .once
+        end
+      end
+
+      context "when `Ractor` API is not supported" do
+        before { hide_const("Ractor") }
```

Comment:
Oh smart! I thought adding JRuby to the CI matrix would catch this but we might as well test it here as well I guess. 👍🏻